enhancements/microshift/generic-device-plugin-for-microshift.md

```diff
@@ -0,0 +1,356 @@
+---
+title: generic-device-plugin-for-microshift
+authors:
+  - pmtk
+reviewers:
+  - "@DanielFroehlich, MicroShift PM"
+  - "@ggiguash, MicroShift contributor"
+  - "@pacevedom, MicroShift contributor"
+  - "@eslutsky, MicroShift contributor"
+  - "@copejon, MicroShift contributor"
+approvers:
+  - "@jerpeter1, MicroShift Staff Engineer"
+api-approvers:
+  - none
+creation-date: 2025-05-13
+last-updated: 2025-05-13
+tracking-link:
+  - https://issues.redhat.com/browse/OCPSTRAT-2031
+---
+
+# Generic Device Plugin for MicroShift
+
+## Summary
+
+[Generic Device Plugin](https://github.com/squat/generic-device-plugin) is a
+straightforward device plugin for devices not needing special initialization
+such as serial, USB, certain video devices. The goal of this enhancement is to
+outline its adoption for MicroShift.
+
+## Motivation
+
+MicroShift is built for edge devices, which commonly use peripherals
+like sensors and adapters.
+Currently, MicroShift does not provide any solution to pass these peripherals
+through to Pods without requiring elevated privileges.
+We want to provide a solution for attaching simple devices (not requiring any special
+initialization) to Pods without potentially compromising on security.
+
+### User Stories
+
+* As a MicroShift device administrator, I want to attach simple devices to Pods
+  without elevating privileges (e.g. mounting `/dev`, adding extraneous capabilities).
+
+### Goals
+
+* Include functionality of
+  [Generic Device Plugin](https://github.com/squat/generic-device-plugin)
+  in MicroShift binary (as a controller)
+
+### Non-Goals
+
+* Create container images
+* Adapt Generic Device Plugin (e.g. create Operator) for OpenShift
+
+## Proposal
+
+* Fork the [Generic Device Plugin](https://github.com/squat/generic-device-plugin)
+  to `github.com/openshift` organization. Generic Device Plugin has Apache 2.0
+  license and forking should not be a problem. Forked repository should include
+  a reference to original repository.
+* Modify fork to make it's more "code import friendly"
+* Import Generic Device Plugin fork into MicroShift binary
+* Integrate the Generic Device Plugin's configuration into MicroShift config
+  and add a toggle option
+
+### Workflow Description
+
+**User** is a human user responsible for setting up and managing Edge Devices.
+
+1. The user enables the Generic Device Plugin using MicroShift configuration
+1. Optionally, the user modifies the configuration of Generic Device Plugin, 
+   for example changes device groups and paths
+1. The user (re)starts MicroShift
+1. MicroShift starts the Generic Device Plugin controller which creates plugin
+   goroutines for each device group.
+1. Plugin goroutines detect the devices and report them with the Kubelet
+1. Kubelet updates its status with new devices
```

Comment:
Not clear, sorry. Can you elaborate?

Reply:
Check now if it's clearer. It's bit of Kubernetes' internal implementation detail